templates/tower-project.yaml

```
@@ -465,30 +479,60 @@ Resources:
               - "s3:GetBucketLocation"
               - "s3:ListBucket"
               - "s3:GetObject"
-              - "s3:PutObject"
-              - "s3:PutObjectTagging"
               - "s3:DeleteObject"
-              - "s3:*MultipartUpload*"
             Resource:
               - !Sub "arn:aws:s3:::${TowerScratch}"
               - !Sub "arn:aws:s3:::${TowerScratch}/*"
+          - Sid: TowerForgeRoleS3WriteAccess
+            Effect: Allow
+            Principal:
+              AWS:
+                - !GetAtt TowerForgeServiceRole.Arn
+                - !GetAtt TowerForgeBatchHeadJobRole.Arn
+                - !GetAtt TowerForgeBatchWorkJobRole.Arn
+            Action:
+              - "s3:PutObject"
+              - "s3:PutObjectTagging"
+              - "s3:*MultipartUpload*"
+            Resource:
+              - !Sub "arn:aws:s3:::${TowerScratch}/work/*"
+              - !Sub "arn:aws:s3:::${TowerScratch}/07_days/*"
+              - !Sub "arn:aws:s3:::${TowerScratch}/14_days/*"
+              - !Sub "arn:aws:s3:::${TowerScratch}/30_days/*"
+              - !Sub "arn:aws:s3:::${TowerScratch}/60_days/*"
+              - !Sub "arn:aws:s3:::${TowerScratch}/90_days/*"
           - !If
             - HasS3ReadWriteAccessArns
-            - Sid: GrantReadWriteAccessToS3ReadWriteAccessArns
+            - Sid: GrantReadDeleteAccessToS3ReadWriteAccessArns
               Effect: Allow
               Principal:
                 AWS: !Ref S3ReadWriteAccessArns
               Action:
                 - "s3:GetObject*"
-                - "s3:PutObject*"
-                - "s3:DeleteObject*"
-                - "s3:*MultipartUpload*"
                 - "s3:ListBucket*"
                 - "s3:GetBucketLocation"
+                - "s3:DeleteObject*"
               Resource:
                 - !Sub "arn:aws:s3:::${TowerScratch}"
                 - !Sub "arn:aws:s3:::${TowerScratch}/*"
             - !Ref AWS::NoValue
+          - !If
+            - HasS3ReadWriteAccessArns
+            - Sid: GrantWriteAccessToS3ReadWriteAccessArns
+              Effect: Allow
+              Principal:
+                AWS: !Ref S3ReadWriteAccessArns
+              Action:
+                - "s3:PutObject*"
+                - "s3:*MultipartUpload*"
+              Resource:
+                - !Sub "arn:aws:s3:::${TowerScratch}/work/*"
```

Same here (see above comment). 

```suggestion
                - !Sub "arn:aws:s3:::${TowerScratch}"
                - !Sub "arn:aws:s3:::${TowerScratch}/work/*"
```